extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -613,16 +643,40 @@ export class AmpAutocomplete extends AMP.BaseElement {
   }
 
   /**
-   * Filter the source data according to the given input and render it in
-   * the results container_.
+   * Display autocomplete suggestions and render it in the results container_.
+   * When client side rendering, filter the source data according to the given opt_input.
+   * @param {?Array<!JsonObject|string>} data
+   * @param {string} opt_input
+   * @return {!Promise}
+   * @private
+   */
+  autocomplete_(data, opt_input = '') {
```

Comment:
This looks better, thanks. For future work: should we add user messaging for the failure cases below?

Reply:
Good idea, that is more in line with how other components behave, esp regarding `data['html']`